link/syscalls.go

```diff
@@ -47,10 +47,6 @@ var haveProgAttach = internal.NewFeatureTest("BPF_PROG_ATTACH", "4.10", func() e
 })
 
 var haveProgAttachReplace = internal.NewFeatureTest("BPF_PROG_ATTACH atomic replacement of MULTI progs", "5.5", func() error {
-	if err := haveProgAttach(); err != nil {
```

Comment:
Ah, this is an interesting case. Here we have to call `haveProgAttach` first since it establishes a pre-condition for the feature test to be successful. If I remember correctly it's so that we can reliably get EBADF from the syscall below.

So I think this needs to stay where it is.

Reply:
I might have to dig into the why myself but why is it okay if it returns `EBADF`?